diff --git a/tests/test-config/valid-config.yaml b/tests/test-config/valid-config.yaml
--- a/tests/test-config/valid-config.yaml
+++ b/tests/test-config/valid-config.yaml
@@ -2,6 +2,12 @@ default_context:
     full_name: "Firstname Lastname"
     email: "firstname.lastname@gmail.com"
     github_username: "example"
+    project:
+        description: "description"
+        tags:
+            - "first"
+            - "second"
+            - "third"
 cookiecutters_dir: "/home/example/some-path-to-templates"
 replay_dir: "/home/example/some-path-to-replay-files"
 abbreviations:
diff --git a/tests/test_get_config.py b/tests/test_get_config.py
--- a/tests/test_get_config.py
+++ b/tests/test_get_config.py
@@ -59,6 +59,10 @@ def test_get_config():
             'full_name': 'Firstname Lastname',
             'email': 'firstname.lastname@gmail.com',
             'github_username': 'example',
+            'project': {
+                'description': 'description',
+                'tags': ['first', 'second', 'third',],
+            },
         },
         'abbreviations': {
             'gh': 'https://github.com/{0}.git',
diff --git a/tests/test_get_user_config.py b/tests/test_get_user_config.py
--- a/tests/test_get_user_config.py
+++ b/tests/test_get_user_config.py
@@ -46,6 +46,10 @@ def custom_config():
             'full_name': 'Firstname Lastname',
             'email': 'firstname.lastname@gmail.com',
             'github_username': 'example',
+            'project': {
+                'description': 'description',
+                'tags': ['first', 'second', 'third',],
+            },
         },
         'cookiecutters_dir': '/home/example/some-path-to-templates',
         'replay_dir': '/home/example/some-path-to-replay-files',
