diff --git a/cookiecutter/config.py b/cookiecutter/config.py
--- a/cookiecutter/config.py
+++ b/cookiecutter/config.py
@@ -45,7 +45,7 @@ def merge_configs(default, overwrite):
         # Make sure to preserve existing items in
         # nested dicts, for example `abbreviations`
         if isinstance(v, dict):
-            new_config[k] = merge_configs(default[k], v)
+            new_config[k] = merge_configs(default.get(k, {}), v)
         else:
             new_config[k] = v
 
